### PR TITLE
Send SprintReport results into Google Sheets

### PR DESCRIPTION
**Pull request by MustafaHaddara**
Swiping from my comments in `README.md`: 

I created a new "Cloud Platform project" [here](https://developers.google.com/sheets/api/quickstart/java). This generated a set of credentials specifically for this project; I can pass these on to you.

The first time you run the application, it'll open in the browser and ask you to authenticate; click ok. 

This will create an entry in the `/tokens` folder. If you need to re-authenticate or change the permissions required by the application, delete the tokens folder to re-auth next time the service starts.

From there, create a spreadsheet with the columns "Committed, Injected, Inflated, Removed, Completed, Incomplete" (in any order, doesn't have to be continuous) and paste the sheet Id into `SheetsTest.java`. The sheets Id can be found in the URL; it'll look something like `17guP0zy95DkAGKhEpExeNUyjMNB_BoJnNIy3yeothMU`

### CODE REVIEW

File: src/main/java/com/ashkan/jira/sheets/SheetsUpdater.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+			String type = loc.getValue();
+
+			int value;
+			switch (type) {
+				case HEADER_COMMITED:   value = report.getCommitted(); break;
+				case HEADER_INJECTED:   value = report.getInjected(); break;
+				case HEADER_INFLATED:   value = report.getInflated(); break;
+				case HEADER_REMOVED:    value = report.getRemoved(); break;
+				case HEADER_COMPLETED:  value = report.getCompletedStoryPoints(); break;
+				case HEADER_INCOMPLETE: value = report.getIncomplete(); break;
```
Comment (MustafaHaddara) [inline src/main/java/com/ashkan/jira/sheets/SheetsUpdater.java:54]: these getters will all need to switch over to the `getXStoryPoints()` variant-- in #2 and #3, we're using these getters for number of tickets.

File: build.gradle [hunk 2]
```diff
@@ -46,4 +46,8 @@ dependencies {
     testImplementation "org.junit.jupiter:junit-jupiter-api:$junitVersion"
     testCompileOnly "org.projectlombok:lombok:$lombokVersion"
     testAnnotationProcessor "org.projectlombok:lombok:$lombokVersion"
+
+    compile 'com.google.api-client:google-api-client:1.30.4'
+    compile "com.google.oauth-client:google-oauth-client-jetty:$googleOAuthClientVersion"
+    compile 'com.google.apis:google-api-services-sheets:v4-rev581-1.25.0'
```
Comment (ashkanent) [inline build.gradle:52]: (nit) let's use `implementation` for all of the above

File: src/main/java/com/ashkan/jira/SheetsTest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package com.ashkan.jira;
+
+import com.ashkan.jira.service.SprintReport;
+import com.ashkan.jira.sheets.SheetsUpdater;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+public class SheetsTest {
+	public static void main(String... args) throws IOException, GeneralSecurityException {
```
Comment (ashkanent) [inline src/main/java/com/ashkan/jira/SheetsTest.java:10]: I think we should avoid two main files, we can call it using spring, maybe in `ClientMain`

File: src/main/java/com/ashkan/jira/SheetsTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+		// Build a new authorized API client service.
+		
+		final String spreadsheetId = "17guP0zy95DkAGKhEpExeNUyjMNB_BoJnNIy3yeothMU";
```
Comment (ashkanent) [inline src/main/java/com/ashkan/jira/SheetsTest.java:13]: we can look into how to optimize this class, I think for sheetId we can rely on env var and read it from `application.propeprties` to hide it here

File: src/main/java/com/ashkan/jira/service/SprintReport.java [hunk 5]
```diff
@@ -11,6 +14,8 @@
 import java.io.StringWriter;
 
 @Component
+@Getter
+@Setter
```
Comment (ashkanent) [inline src/main/java/com/ashkan/jira/service/SprintReport.java:18]:
```suggestion
@Data
```

File: src/main/java/com/ashkan/jira/sheets/HeaderDetector.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@RequiredArgsConstructor
+public class HeaderDetector {
+	public static final String HEADER_COMMITED   = "Committed";
```
Comment (ashkanent) [inline src/main/java/com/ashkan/jira/sheets/HeaderDetector.java:19]: *COMMITTED

**Pull request merged**